include/teak/types.h

```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: Zlib
+//
+// Copyright (c) 2023 Antonio Niño Díaz
+
+#ifndef LIBTEAK_TYPES_H__
+#define LIBTEAK_TYPES_H__
+
+#include <stdint.h>
+
+#define BIT(n) (1 << (n))
+
+typedef volatile uint16_t vu16;
```

Comment:
If we're going to add the common shorthands for types here, we might as well add all the `u/s/vu/vs` types from `ndstypes.h`.

Reply:
Done.